packages/tpetra/core/src/Tpetra_CrsMatrix_def.hpp

```diff
@@ -764,6 +764,7 @@ namespace Tpetra {
       Kokkos::resize(newvals,vals.extent(0));
       Kokkos::deep_copy(newvals,vals);
       k_values1D_ = newvals;   
+      this->fillComplete(source.getDomainMap(),source.getRangeMap());
```

Comment:
1. Getting `vals` will only work if the input matrix is static profile.
2. The input CrsMatrix may not necessarily be fill complete.
```c++
if (source.isFillComplete ()) {
  this->fillComplete (source.getDomainMap (), source.getRangeMap ());
}
```

Reply:
Fixed.